### PR TITLE
[SMALLFIX] Set rules for maven-enforce-plugin to check Java and hadoop version (with YARN)

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -376,6 +377,28 @@
...
+        <executions>
+          <execution>
+            <id>enforce-versions</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireJavaVersion>
+                  <version>${java.version}</version>
```
Comment (calvinjia) [inline pom.xml]: Does this mean it won't work with anything other than java 6?

File: pom.xml [hunk 2]
```diff
@@ -329,6 +329,7 @@
         </plugin>
       </plugins>
     </pluginManagement>
+
```
Comment (aaudiber) [inline pom.xml]: +1

File: pom.xml [hunk 3]
```diff
@@ -376,6 +377,28 @@
...
+        <version>1.4</version>
+        <executions>
+          <execution>
+            <id>enforce-versions</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireJavaVersion>
```
Comment (aaudiber) [inline pom.xml]: Could this be added to the `<rules>` section where we enforce the hadoop version?

File: pom.xml (hunk 1 above)
Comment (apc999) [inline pom.xml]: no it means working with anything above java 6

File: pom.xml (hunk 3 above)
Comment (apc999) [inline pom.xml]: unfortunately, no. it only comes with java version and maven version, not hadoop

File: pom.xml (hunk 1 above)
Comment (apc999) [inline pom.xml]: actually I was lying. this indeed requires java 6. Fixing it .

Comment (apc999) [inline pom.xml]: oops, I was still right. here is the doc about the version syntax:
https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html

File: integration/yarn/pom.xml [hunk 4]
```diff
@@ -47,4 +47,33 @@
...
+          <execution>
+            <id>enforce-versions</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireProperty>
+                  <property>hadoop.version</property>
+                  <regex>[^12]\..*|2\.[4-9]\..*</regex>
```
Comment (calvinjia) [inline integration/yarn/pom.xml]: Will this regex be happy with cdh/hdp distributions?

File: pom.xml (hunk 3 above)
Comment (aaudiber) [inline pom.xml]: Could it just be one section that looks like this? Then if we upgrade the maven-enforcer-plugin we only need to change on place.
```
<rules>
  <!-- Enforce Hadoop version -->
  <requireProperty>
    <property>hadoop.version</property>
    <regex>[^12]\..*|2\.[4-9]\..*</regex>
    <regexMessage>
      hadoop.version must be 2.4.0 or later to build yarn integration.
    </regexMessage>
  </requireProperty>

  <!-- Enforce Java version -->
  <requireJavaVersion>
    <version>${java.version}</version>
  </requireJavaVersion>
</rules>```

File: integration/yarn/pom.xml (hunk 4 above)
Comment (apc999) [inline integration/yarn/pom.xml]: good point. I will update this regex.

File: pom.xml (hunk 3 above)
Comment (apc999) [inline pom.xml]: emm. the problem is we only want to enforce hadoop version (>=2.4.0) in pom of  `integration-yarn` module, not the parent pom. If yarn module is not activated, hadoop version could be 1.0.4 or later

File: integration/yarn/pom.xml (hunk 4 above)
Comment (apc999) [inline integration/yarn/pom.xml]: it works with cdh or hdp which simply adds suffixes to the main hadoop version.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**